File: examples/python/input/datepicker.py
#!/usr/bin/env python3
"""Calendar month grid for picking a date.

    make run-example EX=python/input/datepicker

Keys: arrows move by day/week, PageUp/Down change the month, Enter picks.
"""

from __future__ import annotations

import datetime

import sparcli as sc


def main() -> None:
    if not sc.input_available():
        sc.alert.warning('Run this example in a real terminal (not piped).')
        return

    # `None` seeds the picker with today's date; returns a datetime.date.
    delivery = sc.datepicker(None, sc.DatePickerOpts(
        prompt='Delivery date', week_start=sc.WeekStart.MONDAY,
        accent=sc.Color.CYAN))
    if delivery is not None:
        sc.println(f'  -> {delivery:%A, %d %B %Y}')

    # Pre-seed a specific start date.
    start = sc.datepicker(datetime.date(2027, 1, 1), sc.DatePickerOpts(
        prompt='Starts at 2027-01-01 (Sunday weeks)',
        week_start=sc.WeekStart.SUNDAY))
    if start is not None:
        sc.println(f'  -> {start:%Y-%m-%d}')


if __name__ == '__main__':
    main()

File: examples/rust/input/datepicker.rs
//! Calendar month grid for picking a date.
//!
//!   cargo run -p sparcli --example input_datepicker
//!
//! Keys: arrows move by day/week, PageUp/Down change the month, Enter picks.

use sparcli::*;

fn main() -> sparcli::Result<()> {
    if !input_available() {
        alert::warning("Run this example in a real terminal (not piped).");
        return Ok(());
    }

    // `None` seeds the picker with today's date.
    if let Some(date) = datepicker(
        None,
        DatePickerOpts::new().prompt("Delivery date").week_start(WeekStart::Monday),
    )? {
        println(
            &format!("  -> {:04}-{:02}-{:02}", date.year, date.month, date.day),
            Style::default(),
        );
    }

    // Pre-seed a specific start date.
    let seed = Date { year: 2027, month: 1, day: 1 };
    if let Some(date) = datepicker(
        Some(seed),
        DatePickerOpts::new()
            .prompt("Starts at 2027-01-01 (Sunday weeks)")
            .week_start(WeekStart::Sunday),
    )? {
        println(
            &format!("  -> {:04}-{:02}-{:02}", date.year, date.month, date.day),
            Style::default(),
        );
    }

    Ok(())
}
